src/actions/cartActions.js

```diff
@@ -45,3 +46,94 @@ export const listCartItems = () => async (dispatch) => {
 		});
 	}
 };
+export const addProductToCart = (new_cart_item) => async (dispatch) => {
+	const newCartProduct = {};
+	const newItemId = nextId();
+	try {
+		dispatch({
+			type: CART_ITEM_ADD_REQUEST,
+			payload: newCartProduct,
+		});
+
+		const cartItemRef = doc(db, "cartItems", newItemId);
+
+		const docSnap = await getDoc(cartItemRef);
+
+		if (docSnap.exists()) {
+			const existItem = docSnap.data();
+			alert(existItem.title + " already in cart");
+			dispatch({
+				type: CART_ITEM_ADD_SUCCESS,
+				payload: existItem,
+			});
+		} else {
+			console.log("No such document!");
+			await setDoc(doc(db, "cartItems", newItemId), {
+				id: newItemId,
+				title: new_cart_item.title,
+				price: new_cart_item.price,
+				image: new_cart_item.image,
+				qtyInCart: 1,
+			});
+			alert(new_cart_item.title + " successfully added");
+
+			dispatch({
+				type: CART_ITEM_ADD_SUCCESS,
+				payload: newCartProduct,
+			});
+		}
+	} catch (error) {
+		alert("Failed To Add " + new_cart_item.title + error);
+		dispatch({
+			type: CART_ITEM_ADD_FAIL,
+			payload:
+				error.response && error.response.data.message
+					? error.response.data.message
+					: error.message,
+		});
+	}
+};
+export const updateCartQty = (cart_item_id, qty) => async (dispatch) => {

```

Comment:
What does mean `QTY`?

Reply:
![image](https://user-images.githubusercontent.com/74628686/166253898-225d9f7c-267a-4b05-9fa7-d4d4656d642d.png)

Reply:
Then let's rename to updateCartCounter or updateCartQuantity. 
QTY isn't abbreviation